File: modules/spring-core/testdata/java/providers/linemarkers/beans/one_bean_two_dependency/TestConfiguration.java
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration
import org.springframework.stereotype.Component;

class FooAOneBeanTwoDependency{}

class FooBOneBeanTwoDependency{}

@Configuration
class TestConfigurationOneBeanTwoDependency {
    @Bean
    FooAOneBeanTwoDependency foo() { return new FooAOneBeanTwoDependency(); }
}

@Component
class FooAOneBeanTwoDependencyComponent1 {
    @Autowired FooAOneBeanTwoDependency foo;
}

@Component
class FooAOneBeanTwoDependencyComponent2 {
    @Autowired FooAOneBeanTwoDependency foo;
}

@Component
class FooBOneBeanTwoDependencyComponent1 {
    @Autowired FooBOneBeanTwoDependency foo;
}

@Component
class FooBOneBeanTwoDependencyComponent2 {
    @Autowired FooBOneBeanTwoDependency foo;
}

File: modules/spring-core/testdata/kotlin/providers/linemarkers/beans/one_bean_two_dependency/TestConfiguration.kt
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.stereotype.Component

internal class FooAOneBeanTwoDependency

internal class FooBOneBeanTwoDependency

@Configuration
internal class TestConfigurationOneBeanTwoDependency {
    @Bean
    fun foo(): FooAOneBeanTwoDependency {
        return FooAOneBeanTwoDependency()
    }
}

@Component
internal class FooAOneBeanTwoDependencyComponent1 {
    @Autowired
    var foo: FooAOneBeanTwoDependency? = null
}

@Component
internal class FooAOneBeanTwoDependencyComponent2 {
    @Autowired
    var foo: FooAOneBeanTwoDependency? = null
}

@Component
internal class FooBOneBeanTwoDependencyComponent1 {
    @Autowired
    var foo: FooBOneBeanTwoDependency? = null
}

@Component
internal class FooBOneBeanTwoDependencyComponent2 {
    @Autowired
    var foo: FooBOneBeanTwoDependency? = null
}